services/src/handlers/wms.rs

```diff
@@ -0,0 +1,91 @@
+use warp::{Filter, http::Response};
+use warp::reply::Reply;
+use std::sync::Arc;
+use tokio::sync::RwLock;
+
+use crate::ogc::wms::request::{WMSRequest, GetCapabilities, GetMap};
+use geoengine_datatypes::raster::Raster2D;
+use geoengine_datatypes::operations::image::{Colorizer, ToPng};
+use crate::error::Error;
+use crate::workflows::registry::WorkflowRegistry;
+
+type WR<T> = Arc<RwLock<T>>;
+
+pub fn wms_handler<T: WorkflowRegistry>(workflow_registry: WR<T>) -> impl Filter<Extract=impl warp::Reply, Error=warp::Rejection> + Clone {
+    warp::get()
+        .and(warp::path!("wms"))
+        .and(warp::query::<WMSRequest>())
+        .and(warp::any().map(move || Arc::clone(&workflow_registry)))
+        .and_then(wms)
+}
+
+// TODO: move into handler once async closures are available?
+async fn wms<T: WorkflowRegistry>(request: WMSRequest, workflow_registry: WR<T>) -> Result<Box<dyn warp::Reply>, warp::Rejection> {
+    // TODO: authentication
+    // TODO: more useful error output than "invalid query string"
+    match request {
+        WMSRequest::GetCapabilities(request) => get_capabilities(&request),
+        WMSRequest::GetMap(request) => get_map(&request, &workflow_registry),
+        // TODO: support other requests
+        _ => Ok(Box::new(warp::http::StatusCode::NOT_IMPLEMENTED.into_response()))
+    }
+}
+
+fn get_capabilities(_request: &GetCapabilities) -> Result<Box<dyn warp::Reply>, warp::Rejection> {
+    // TODO: implement
```

Comment:
Is there a way to provide a minimal implementation of that?
We can't provide a list of layers, but maybe a minimum set of features for the WMS.

Reply:
What would be a minimal implementation? A hard-coded string that is returned? The GetCapabilities output is quite verbose: https://maps6.geosolutionsgroup.com/geoserver/osm/wms?SERVICE=WMS&VERSION=1.1.1&REQUEST=GetCapabilities

Reply:
I guess hardcoded makes sense since we would need to know what we implemented programmatically.
We can skip the layers and just show the requests and formats that we've implemented, so `GetMap` for now and `image/png`.

Reply:
2274712368246cce10c2bcde171eefd51f9eb3bf